src/TraverseTracks.cpp

```diff
@@ -0,0 +1,154 @@
+#include "TraverseTracks.h"
+
+/**
+ * @brief Constructor for the TravseSegments class assigns the TrackGenerator
+ *        and pulls relevant information from it.
+ */
+TraverseTracks::TraverseTracks(TrackGenerator* track_generator) {
+
+  /* Save the track generator */
+  _track_generator = track_generator;
+
+  /* Determine the type of segment formation used */
+  _segment_formation = track_generator->getSegmentFormation();
+}
+
+
+/**
+ * @brief Destructor for TraverseTracks
+ */
+TraverseTracks::~TraverseTracks() {
+}
+
+
+/**
+ * @breif Loops over Tracks, applying the provided kernel to all segments and
+ *        the functionality described in onTrack(...) to all Tracks.
+ * @details The segment formation method imported from the TrackGenerator
+ *          during construction is used to redirect to the appropriate looping
+ *          scheme.
+ * @param kernel The MOCKernel to apply to all segments
+ */
+void TraverseTracks::loopOverTracks(MOCKernel* kernel) {
+  switch (_segment_formation) {
```

Comment:
is this `switch-case` statement really necessary? Perhaps it's just a stub for the future.